modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -795,4 +801,72 @@ private void consumeHttpRequestInfo() {
             onHttpRequest.run();
         }
     }
+
+    OptionalLong getRegister(String blobPath, String containerPath, String blobKey) {
+        try {
+            return SocketAccess.doPrivilegedException(() -> {
+                final var blobClient = getAzureBlobServiceClientClient().getSyncClient()
+                    .getBlobContainerClient(container)
+                    .getBlobClient(blobPath);
+                return OptionalLong.of(
+                    BlobContainerUtils.getRegisterUsingConsistentRead(blobClient.downloadContent().toStream(), containerPath, blobKey)
+                );
+            });
+        } catch (Exception e) {
+            if (Throwables.getRootCause(e)instanceof BlobStorageException blobStorageException
+                && blobStorageException.getStatusCode() == RestStatus.NOT_FOUND.getStatus()) {
+                return OptionalLong.empty();
+            }
+            throw e;
+        }
+    }
+
+    OptionalLong compareAndExchangeRegister(String blobPath, String containerPath, String blobKey, long expected, long updated) {
+        try {
+            return SocketAccess.doPrivilegedException(() -> {
+
+                final var blobClient = getAzureBlobServiceClientClient().getSyncClient()
+                    .getBlobContainerClient(container)
+                    .getBlobClient(blobPath);
+
+                if (blobClient.exists()) {
+                    final var leaseClient = new BlobLeaseClientBuilder().blobClient(blobClient).buildClient();
+                    final var leaseId = leaseClient.acquireLease(60);
+                    try {
+                        final var currentValue = getRegister(blobPath, containerPath, blobKey);
```

Comment:
I think we should get the register using the acquired lease? See https://learn.microsoft.com/en-us/rest/api/storageservices/get-blob?tabs=azure-ad#request-headers

Reply:
ah yes you're right

Reply:
done in [a03a836](https://github.com/elastic/elasticsearch/pull/94276/commits/a03a836f73c2736500cadc7ec13ebd27070feb73)